### PR TITLE
[Refactor] Make coordinator class more organized

### LINKED ISSUE #384: [Improvement] Make coordinator class more organized
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-uniffle/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Currently, the class in coordinator package are all in single folder. We should make coordinator class more organized, and split them into different folders by their correlation.

### How should we improve?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by smallzhongfeng**
### What changes were proposed in this pull request?

Split coordinator class into different folders by their correlation.

### Why are the changes needed?

To resolve #384 

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Origin tests.

### CODE REVIEW

**Comment by smallzhongfeng** [comment]
```
Error:  High: org.apache.uniffle.coordinator.ApplicationManager.REMOTE_PATH_SCHEMA is a mutable collection [org.apache.uniffle.coordinator.ApplicationManager] At ApplicationManager.java:[line 53] MS_MUTABLE_COLLECTION
```
Waht could I resolve this?

**Comment by zuston** [comment]
> ```
> Error:  High: org.apache.uniffle.coordinator.ApplicationManager.REMOTE_PATH_SCHEMA is a mutable collection [org.apache.uniffle.coordinator.ApplicationManager] At ApplicationManager.java:[line 53] MS_MUTABLE_COLLECTION
> ```
> 
> Waht could I resolve this? I haven't met before. @zuston

Make it `final?`

**Comment by zuston** [comment]
1. Should we put `checker` and `access` folders together? Because they are related.
2. Another concern is the package renaming, which will make this PR incompatible with original config, like ‘org.apache.uniffle.coordinator.checker.AccessClusterLoadChecker’. Further more, I suggest that we'd better not to use the package name as the config of strategy in the future @jerqi Do you think so?

**Comment by jerqi** [comment]
> 1. Should we put `checker` and `access` folders together? Because they are related.
> 2. Another concern is the package renaming, which will make this PR incompatible with original config, like ‘org.apache.uniffle.coordinator.checker.AccessClusterLoadChecker’. Further more, I suggest that we'd better not to use the package name as the config of strategy in the future @jerqi Do you think so?

1. move checker directory to access directory?
2. Em..it's ok for me actually. We can add a document to explain how migrate to higher version if we make some imcompatible changes. Or we add short name for some configs. Or we introduce the deprecated configuration option.

**Comment by zuston** [comment]
+1. Short name or concrete strategy name will be better.

**Comment by jerqi** [comment]
> +1. Short name or concrete strategy name will be better.

If we use short name, this pr will also be imcompatible change.

**Comment by zuston** [comment]
> > +1. Short name or concrete strategy name will be better.
> 
> If we use short name, this pr will also be incompatible change.

Yes. The refactor process will always make incompatible change. I think it could be optimized in later PRs, currently the strategies are bound to package name, it looks unreasonable. 

We should reach a consensus that this way is no longer supported for new PRs. Do you think so? 

By the way, I just think more when reviewing this PR.

**Comment by jerqi** [comment]
cc @kaijchen 

> > > +1. Short name or concrete strategy name will be better.
> > 
> > 
> > If we use short name, this pr will also be incompatible change.
> 
> Yes. The refactor process will always make incompatible change. I think it could be optimized in later PRs, currently the strategies are bound to package name, it looks unreasonable.
> 
> We should reach a consensus that this way is no longer supported for new PRs. Do you think so?
> 
> By the way, I just think more when reviewing this PR.

I'm ok for that it is bound to package. Other systems have also similar style.

**Comment by kaijchen** [comment]
@smallzhongfeng is this a refactor? If so, please change `[Improvement]` to `[Refactor]` in title.

**Comment by smallzhongfeng** [comment]
> Yes. The refactor process will always make incompatible change. I think it could be optimized in later PRs, currently the strategies are bound to package name, it looks unreasonable.
> 
> We should reach a consensus that this way is no longer supported for new PRs. Do you think so?
> 
> By the way, I just think more when reviewing this PR.

I make it final, but it doesn't work.

**Comment by smallzhongfeng** [comment]
> * move checker directory to access directory?
> * Em..it's ok for me actually. We can add a document to explain how migrate to higher version if we make some imcompatible changes. Or we add short name for some configs. Or we introduce the deprecated configuration option.

1. I will move it.
2. What should I do? Do I need to add a migration document?

**Comment by smallzhongfeng** [comment]
> @smallzhongfeng is this a refactor? If so, please change `[Improvement]` to `[Refactor]` in title.

I think so. @jerqi @zuston WDYT?

**Comment by jerqi** [comment]
> > * move checker directory to access directory?
> > * Em..it's ok for me actually. We can add a document to explain how migrate to higher version if we make some imcompatible changes. Or we add short name for some configs. Or we introduce the deprecated configuration option.
> 
> 1. I will move it.
> 2. What should I do? Do I need to add a migration document?

2. Yes, if you have imcompatible feature, you should add migration document, you can refer to https://github.com/apache/spark/blob/master/docs/core-migration-guide.md

**Comment by smallzhongfeng** [comment]
> ```
> Error:  High: org.apache.uniffle.coordinator.ApplicationManager.REMOTE_PATH_SCHEMA is a mutable collection [org.apache.uniffle.coordinator.ApplicationManager] At ApplicationManager.java:[line 53] MS_MUTABLE_COLLECTION
> ```
> 
> Waht could I resolve this? I haven't met before. @zuston

What is the problem? I haven't met it before.@jerqi

File: docs/coordinator-migration-guide.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+---
+layout: page
```
Comment (jerqi) [inline docs/coordinator-migration-guide.md:2]: I feel that it's enough to have an entire document. We don't need to separate it into modules, such as coordinator, server, client.

Comment (smallzhongfeng) [reply docs/coordinator-migration-guide.md:2]: It's ok for me, updated.

Comment (jerqi) [reply docs/coordinator-migration-guide.md:2]: The file name is still `coordinator-migration-guide.md`.

**Comment by jerqi** [comment]
LGTM, @zuston Do you have another suggestion? Let @zuston approve and merge this pr.

**Review by jerqi — APPROVED** [review]

**Review by zuston — APPROVED** [review]
LGTM

**Comment by zuston** [comment]
Thanks @jerqi @smallzhongfeng Merged.

**Comment by smallzhongfeng** [comment]
Belated thanks. @jerqi @zuston

**Pull request merged**